x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -63,300 +129,493 @@ export const DeploymentSetup: FC<DeploymentSetupProps> = ({
+    const vcpuRange = deploymentParamsMapper.getVCPURange(config.vCPUUsage);
+
+    if (cloudInfo.isCloud && cloudInfo.isMlAutoscalingEnabled) {
+      // Running in cloud with ML autoscaling enabled
+      if (config.adaptiveResources) {
+        // With adaptive resources
+        switch (config.vCPUUsage) {
+          case 'low':
+            return i18n.translate(
+              'xpack.ml.trainedModels.modelsList.startDeployment.cloudAutoscaling.lowCpuAdaptiveHelp',
+              {
+                defaultMessage:
+                  'This level limits resources to the minimum required for ELSER to run if supported by your Cloud console selection. It may not be sufficient for a production application.',
+              }
+            );
+          case 'medium':
+            return i18n.translate(
+              'xpack.ml.trainedModels.modelsList.startDeployment.cloudAutoscaling.mediumCpuAdaptiveHelp',
+              {
+                defaultMessage:
+                  'Your model will scale up to a maximum of {maxVCPUs} vCPUs. Even if the Cloud console provides more, the model will not exceed {maxVCPUs}, leaving additional resources for other models.',
+                values: { maxVCPUs: vcpuRange.max },
+              }
+            );
+          case 'high':
+            return i18n.translate(
+              'xpack.ml.trainedModels.modelsList.startDeployment.cloudAutoscaling.highCpuAdaptiveHelp',
+              {
+                defaultMessage:
+                  'Your model may scale up to the maximum number of vCPUs available for this deployment from the Cloud console if needed. If the maximum is {minVCPUs} vCPUs or fewer, this level is equivalent to the medium level.',
+                // use an upper bound of medium level
+                values: { minVCPUs: vcpuRange.min - 1 },
+              }
+            );
         }
-        description={
-          <FormattedMessage
-            id="xpack.ml.trainedModels.modelsList.startDeployment.deploymentIdHelp"
-            defaultMessage="Specify unique identifier for the model deployment."
-          />
+      } else {
+        // Without adaptive resources
+        switch (config.vCPUUsage) {
+          case 'low':
+            return i18n.translate(
+              'xpack.ml.trainedModels.modelsList.startDeployment.cloudAutoscaling.lowCpuStaticHelp',
+              {
+                defaultMessage:
+                  'This level sets resources to the minimum required for ELSER to run if supported by your Cloud console selection. It may not be sufficient for a production application.',
+              }
+            );
+          case 'medium':
+            return i18n.translate(
+              'xpack.ml.trainedModels.modelsList.startDeployment.cloudAutoscaling.mediumCpuStaticHelp',
+              {
+                defaultMessage:
+                  'Your model will consume {staticVCPUs} vCPUs consistently, if provided by the Cloud console.',
```

Comment:
A bit longer but it may be a bit more unambiguous, too.

```suggestion
                  'Your model will consume {staticVCPUs} vCPUs, even when not in use, if provided by the Cloud console.',
```

Reply:
Updated in Updated in c3077e11b02d01f3be0acf409a86f4fc0cbb4a43